host/greatfet/utils.py

```diff
@@ -1,13 +1,160 @@
 #
 # This file is part of GreatFET
 #
+"""
+    Utilities that help in wirting simple scripts for GreatFET.
+"""
 
 from __future__ import print_function
 
+import sys
+import errno
+import argparse
+
+from greatfet import GreatFET
+from pygreat.errors import DeviceNotFoundError
+
 def log_silent(string, end=None):
     """Silently discards all log data, but provides our logging interface."""
     pass
 
 def log_verbose(string, end="\n"):
     """Prints all logging data to the screen."""
     print(string, end=end)
+
+
+class GreatFETArgumentParser(argparse.ArgumentParser):
+    """ Convenience-extended argument parser for GreatFET. """
+
+    """ Serial number expected from a device in DFU. """
+    DFU_STUB_SERIAL = "dfu_flash_stub"
+
+    def __init__(self, *args, **kwargs):
+        """ Sets up a GreatFET-specialized argument parser.
+
+        Additional keyword arguments:
+            dfu -- If set to True, DFU-reglated arguemnts will be provided.
+            raise_device_find_failures -- If set to True, this will throw a DeviceNotFoundError
+                instead of quitting if no device is present.
+        """
+            
+        # Determine if we should provide DFU arguments.
+        if 'dfu' in kwargs:
+            self.supports_dfu = kwargs['dfu']
+            del kwargs['dfu']
+        else:
+            self.supports_dfu = False
```

Comment:
Consider using [`pop()`](https://docs.python.org/2/library/stdtypes.html#dict.pop) for these instead:
`self.supports_dfu = kwargs.pop('dfu', False)`